hack/kserve_migration/kserve_migration.sh

```diff
@@ -0,0 +1,236 @@
+#!/bin/bash
+# Usage: kserve_migration.sh
+
+set -o errexit
+set -o errtrace
+
+export KSERVE_VERSION=v0.7.0-rc0
+export CONFIG_DIR="config"
+export ISVC_CONFIG_DIR="${CONFIG_DIR}/isvc"
+export KSVC_CONFIG_DIR="${CONFIG_DIR}/ksvc"
+
+CLEAN_KFSERVING="true"
+
+# custom logger
+log() {
+    level=$1
+    msg=$2
+    ts=$(date -u +'%F %T')
+    echo "${ts} [${level}] $msg"
+}
+
+# Validates whether controller manager and models web app service running 
+# on this machine for the given namespace or not.
+isControllerRunning() {
+    namespace=$1
+    prefix="kserve"
+    if [ "${namespace}" == "kfserving-system" ]; then
+        prefix="kfserving"
+    fi
+    svc_names=$(kubectl get svc -n $namespace -o jsonpath='{.items[*].metadata.name}')
+    for svc_name in "${prefix}-controller-manager-metrics-service" \
+                    "${prefix}-controller-manager-service" \
+                    "${prefix}-models-web-app" \
+                    "${prefix}-webhook-server-service"; do
+        if [ ! -z "${svc_names##*$svc_name*}" ]; then
+            log ERROR "${prefix} controller services are not installed completely."
+            exit 1;
+        fi
+    done
+    po_names=$(kubectl get po -n $namespace -o jsonpath='{.items[*].metadata.name}')
+    for po_name in "${prefix}-controller-manager" "${prefix}-models-web-app"; do
+        if [ ! -z "${po_names##*$po_name*}" ]; then
+            log ERROR "${prefix} controller services are not installed completely."
+            exit 1;
+        fi
+    done
```

Comment:
I think we do not necessarily need to check web app, for our use we do not run web app which is an optional component.

Reply:
Ok @yuzisun . I will remove web app from validation.